apps/devhub/views.py

```diff
@@ -831,6 +831,12 @@ def json_upload_detail(request, upload, addon_slug=None):
     if result['validation']:
         try:
             pkg = parse_addon(upload, addon=addon)
+            if (pkg.get('is_experiment', False) and
+                    not acl.action_allowed(request, 'Experiments', 'submit')):
```

Comment:
We don't always have access to the `request` in `parse_addon`, as it's used in quite a few places, and some of them are not in the request flow (eg we use it for parsing automatically fetched langpacks using crons).

Also, `parse_addon` isn't concerned with user permissions at all, which should be checked in the view I believe. I guess we could have `check_addon` taking a request, and it would check the user permission, and then in turn call the `parse_addon`? And then the API would use `check_addon` instead?

Experiments are a very specific use case: "special" extensions that are only submitted by Mozilla, and I think we'll have some more special extension types in the future. We might want to have something better/cleaner in place to cope with those, but one step at a time ;)

Anyway, I would push back on this comment, and ask your thoughts on doing the user permission check in the API view like it is here (in another PR? the same?). What do you think @mstriemer ?